src/picknik_ur_site_config/objectives/push_button_ml.xml

```diff
@@ -1,94 +1,68 @@
         </Control>
-    </BehaviorTree>
-    <BehaviorTree ID="CloseGripper">
-        <Control ID="Sequence" name="close_gripper_main">
-            <Action ID="MoveGripperAction"
-                gripper_command_action_name="/robotiq_gripper_controller/gripper_cmd"
-                position="0.7929" />
+      </Decorator>
+      <Action ID="GetGraspableObjectsFromMasks3D" point_cloud="{point_cloud}" masks3d="{masks3d}" base_frame="world" plane_inlier_threshold="0.02" minimum_face_area="0.000625" face_separation_threshold="0.03" graspable_objects="{graspable_objects}"/>
+      <Decorator ID="ForceSuccess">
+        <Decorator ID="ForEachGraspableObject" vector_in="{graspable_objects}" out="{button}">
+          <Action ID="AlwaysFailure"/>
+        </Decorator>
+      </Decorator>
+      <Action ID="ExtractGraspableObjectPose" graspable_object="{button}" pose="{button_pose}"/>
+      <SubTree ID="CloseGripper" _collapsed="true"/>
+      <Action ID="SaveCurrentState" saved_robot_state="{initial_robot_state}"/>
+      <Control ID="Sequence">
+        <Action ID="InitializeMTCTask" task_id="push_button_ml" controller_names="/joint_trajectory_controller" task="{push_along_axis_task}"/>
+        <Action ID="SetupMTCCurrentState" task="{push_along_axis_task}"/>
+        <Action ID="SetupMTCMoveToPose" ik_frame="grasp_link" planning_group_name="manipulator" target_pose="{button_pose}" task="{push_along_axis_task}" use_all_planners="false"/>
+        <Action ID="SetupMTCUpdateGroupCollisionRule" name="AllowGripperCollisionWithOctomap" parameters="{parameters}" task="{push_along_axis_task}"/>
+        <Action ID="SetupMTCMoveAlongFrameAxis" task="{push_along_axis_task}" hand_frame="grasp_link" axis_frame="grasp_link" axis_x="0.0" axis_y="0.0" axis_z="1.0" max_distance="0.2" min_distance="0.05" planning_group_name="manipulator" velocity_scale="0.05" acceleration_scale="0.05"/>
+        <Action ID="SetupMTCUpdateGroupCollisionRule" name="ForbidGripperCollisionWithOctomap" parameters="{parameters}" task="{push_along_axis_task}"/>
+        <Action ID="PlanMTCTask" solution="{full_push_along_axis_solution}" task="{push_along_axis_task}"/>
+        <Action ID="SplitMTCSolution" solution_in="{full_push_along_axis_solution}" index="3" solution_out_1="{move_to_approach_solution}" solution_out_2="{push_solution}"/>
+        <Action ID="WaitForUserTrajectoryApproval" solution="{full_push_along_axis_solution}"/>
+        <Action ID="ExecuteMTCTask" solution="{move_to_approach_solution}"/>
+        <Action ID="SaveCurrentState" saved_robot_state="{pre_approach_robot_state}"/>
+        <Control ID="Parallel" success_count="1" failure_count="1">
+          <Action ID="ForceExceedsThreshold" wrench_topic_name="/force_torque_sensor_broadcaster/wrench" hand_frame_name="grasp_link" wrench_frame_name="tool0" minimum_consecutive_wrench_values="10" force_threshold="10.0"/>
```

Comment:
This will be another situation where the regular hardware config and Gazebo configs will need to differ due to different force thresholds. Make sure this is the case.

Reply:
Will need to test this on hardware to ensure subtree works correctly but this should be good now!